File: Blog.Core/TableFilters/TableSortingModel.cs
﻿// <copyright file="TableSortingModel.cs" company="Blog">
// Copyright (c) Blog. All rights reserved.
// </copyright>

namespace Blog.Core.TableFilters
{
    /// <summary>
    /// Table sorting model.
    /// </summary>
    public class TableSortingModel
    {
        /// <summary>
        /// Gets or sets column.
        /// </summary>
        public int Column { get; set; }

        /// <summary>
        /// Gets or sets dir.
        /// </summary>
        public string Dir { get; set; }
    }
}

File: Shared/Blog.FSharp.Core/TableFilters/TableSortingModel.fs
namespace Blog.FSharp.Core.TableFilters

/// <summary>
/// Represents a sorting instruction for a table.
/// </summary>
[<CLIMutable>]
type TableSortingModel = {
    /// <summary>
    /// Index of the column to sort by (zero-based).
    /// </summary>
    Column : int

    /// <summary>
    /// Sorting direction ("asc" or "desc").
    /// </summary>
    Dir : string
}